src/main/scala/rise/core/substitute.scala

```diff
@@ -33,6 +33,7 @@ object substitute {
             case Lambda(x, b) =>
               // See https://www.cs.cornell.edu/courses/cs3110/2019sp/textbook/interp/subst_lambda.html
               if (x =~= `for`) return_(e)
+              // FIXME: should be !(FV(expression) contains x)
```

Comment:
Can you please create an issue for this and link it from here

Reply:
https://github.com/rise-lang/shine/issues/164